aws/credentials/processcreds/provider.go

```diff
@@ -0,0 +1,421 @@
+    creds := credentials.NewCredentials(&processcreds.ProcessProvider{
+        Command:    myCommand,
+        Timeout:    time.Duration(2) * time.Minute,
+        MaxBufSize: 500,
+    })
+*/
+package processcreds
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+)
+
+const (
+	// ProviderName is the name this credentials provider will label any
+	// returned credentials Value with.
+	ProviderName = `ProcessProvider`
+
+	// ErrCodeProcessProviderParse error parsing process output
+	ErrCodeProcessProviderParse = "ProcessProviderParseError"
+
+	// ErrCodeProcessProviderVersion version error in output
+	ErrCodeProcessProviderVersion = "ProcessProviderVersionError"
+
+	// ErrCodeProcessProviderRequired required attribute missing in output
+	ErrCodeProcessProviderRequired = "ProcessProviderRequiredError"
+
+	// ErrCodeProcessProviderExecution execution of command failed
+	ErrCodeProcessProviderExecution = "ProcessProviderExecutionError"
+
+	// ErrMsgProcessProviderTimeout process took longer than allowed
+	ErrMsgProcessProviderTimeout = "credential process timed out"
+
+	// ErrMsgProcessProviderNoKill process could not be killed
+	ErrMsgProcessProviderNoKill = "unable to kill process"
+
+	// ErrMsgProcessProviderProcess process error
+	ErrMsgProcessProviderProcess = "error in credential_process"
+
+	// ErrMsgProcessProviderParse problem parsing output
+	ErrMsgProcessProviderParse = "parse failed of credential_process output"
+
+	// ErrMsgProcessProviderVersion version error in output
+	ErrMsgProcessProviderVersion = "wrong version in process output (not 1)"
+
+	// ErrMsgProcessProviderMissKey missing access key id in output
+	ErrMsgProcessProviderMissKey = "missing AccessKeyId in process output"
+
+	// ErrMsgProcessProviderMissSecret missing secret acess key in output
+	ErrMsgProcessProviderMissSecret = "missing SecretAccessKey in process output"
+
+	// ErrMsgProcessProviderPrepareCmd prepare of command failed
+	ErrMsgProcessProviderPrepareCmd = "failed to prepare command"
+
+	// ErrMsgProcessProviderFewArgs not enough args
+	ErrMsgProcessProviderFewArgs = "not enough args"
+
+	// ErrMsgProcessProviderEmptyCmd command must not be empty
+	ErrMsgProcessProviderEmptyCmd = "command must not be empty"
+
+	// ErrMsgProcessProviderPipe failed to initialize pipe
+	ErrMsgProcessProviderPipe = "failed to initialize pipe"
+
+	// GoOSWindowsKey for identifying Windows OS
+	GoOSWindowsKey = "windows"
+
+	// DefaultDuration is the default amount of time in minutes that the
+	// credentials will be valid for.
+	DefaultDuration = time.Duration(15) * time.Minute
+
+	// DefaultInitialBufSize size for initial buffer.
+	DefaultInitialBufSize = 200
+
+	// DefaultMaxBufSize limits buffer size from growing to an enormous
+	// amount due to a faulty process.
+	DefaultMaxBufSize = 512
```

Comment:
What do you think about merging `DefaultMaxBufSize` and `DefaultInitialBufSize` together into a `DefaultBuffSize=512`? The difference between the two values are fairly minor.

Reply:
Agreed.